platforms/nuttx/NuttX/CMakeLists.txt

```
@@ -180,10 +179,48 @@ function(add_nuttx_dir nuttx_lib nuttx_lib_dir kernel extra target)
 	add_dependencies(nuttx_${nuttx_lib} nuttx_${nuttx_lib}_build)
 endfunction()
 
+# helper for drivers with segger_sysview patches
+function(add_nuttx_dir_patched nuttx_lib nuttx_lib_dir kernel extra target)
```

Maybe a bit more specific naming? I had to dig in to spot the segger differences.

```suggestion
function(add_nuttx_dir_segger_patched nuttx_lib nuttx_lib_dir kernel extra target)
```